phpBB/includes/acp/acp_search.php

```diff
@@ -76,41 +133,39 @@ function settings($id, $mode)
 			'search_store_results'			=> 'integer',
 		];
 
-		$search = null;
 		$search_options = '';
 
-		foreach ($search_types as $search)
+		foreach ($this->search_backend_collection as $search)
 		{
 			// Only show available search backends
 			if ($search->is_available())
 			{
 				$name = $search->get_name();
-				$type = get_class($search);
+				$type = $search->get_type();
 
-				$selected = ($config['search_type'] == $type) ? ' selected="selected"' : '';
+				$selected = ($this->config['search_type'] === $type) ? ' selected="selected"' : '';
 				$identifier = substr($type, strrpos($type, '\\') + 1);
 				$search_options .= "<option value=\"$type\"$selected data-toggle-setting=\"#search_{$identifier}_settings\">$name</option>";
```

Comment:
Probably not an issue here, but since the topic is "rewrite etc... " I would go to remove the HTML in PHP and use TWIG instead  for this.

Reply:
Well, my idea was to rewrite the indexing part, I haven't touch the settings method

Reply:
Of course, just a consideration since TWIG offers us the opportunity.

Reply:
Well, that can be rewritten in the future, when someone do a proper way to handle that kind of forms

Reply:
Do you mean a PHP function that handles forms like `options``?

Reply:
No, something like symfony forms for all forms in acp & ucp

Reply:
This is beyond my expectations, I know for a fact that OPTION-like forms can be easily replaced at this stage of phpBB as we use them a lot in extensions.

Reply:
Certainly outside the scope of this PR though. ;)